katello/katello-change-hostname

```diff
@@ -0,0 +1,190 @@
+    options[:scenario] = scenario
+  end
+
+  opt.on("-d","--disable-system-checks","runs the installer with --disable-system-checks") do |system_check|
+    options[:system_check] = true
+  end
+
+  opt.on("-l","--ssl-dir SSL_DIR","specify the ssl cert directory to replace (defaults to #{DEFAULT_SSL_DIR})") do |ssl_dir|
+    options[:ssl_dir] = ssl_dir
+  end
+
+ opt.on("-h","--help","help") do
+    puts opt_parser
+    exit
+  end
+end
+opt_parser.parse!
+
+raise 'Must run as root' unless Process.uid == 0
+
+if options[:scenario] == "katello"
+  proxy = "Smart Proxy"
+else
+  proxy = "Capsule"
+end
+
+if ARGV[0] && ARGV.count >= 1
+  new_hostname = ARGV[0]
+else
+  puts opt_parser
+  exit
+end
+
+STDOUT.puts("\nA hostname change requires a manual change in the program. " \
+	     "Please log into UI and rename the default #{proxy} (under Infrastructure -> #{proxy}) " \
+             "to #{new_hostname} using both the name and url fields. " \
+             "After this script completes, you will have to re-register any capsules and #{options[:scenario]} clients. " \
+             "All services will be stopped and started during this process. This script will modify your system!")
+STDOUT.print("\nHave you made this necessary change and want to proceed with changing your hostname to #{new_hostname}? [y/n]")
+response = yesno
+STDOUT.puts ""
+
+unless response
+  STDOUT.puts "Hostname change aborted, no changes have been made to your system"
+  exit
+end
+
+# Get the hostname from your system
+old_hostname = get_hostname
+
+if is_rhel_version?("6")
+  STDOUT.puts "updating hostname in /etc/sysconfig/network"
+  STDOUT.puts("sed -i -e 's/#{old_hostname}/#{new_hostname}/g' /etc/sysconfig/network")
+  `sed -i -e 's/#{old_hostname}/#{new_hostname}/g' /etc/sysconfig/network`
+  STDOUT.puts "setting hostname"
+  `hostname #{new_hostname}`
+elsif is_rhel_version?("7")
+  STDOUT.puts "updating hostname in /etc/hostname"
+  `sed -i -e 's/#{old_hostname}/#{new_hostname}/g' /etc/hostname`
+  STDOUT.puts "setting hostname"
+  `hostnamectl set-hostname #{new_hostname}`
+else
+  STDOUT.puts "couldn't find RHEL version: #{RHEL_VERSION} - exiting, no changes have been made to your system"
+  exit
+end
+
+STDOUT.puts "checking if hostname was changed"
+if get_hostname != new_hostname
+  STDOUT.puts "the new hostname was not changed successfully, exiting script"
+  exit
+end
+
+STDOUT.puts "stopping services"
+`katello-service stop`
+
+STDOUT.puts "deleting old certs"
+`
+rm -rf /etc/pki/katello{,.bak}
+rm -rf /etc/pki/katello-certs-tools{,.bak}
+rm -rf /etc/candlepin/certs/amqp{,.bak}
+rm -rf /etc/foreman-proxy/*ssl*
+rm -rf /etc/foreman/old-certs
+rm -rf /etc/foreman/*.pem
+rm -rf /var/lib/puppet/ssl
+rm -rf /etc/pki/katello/nssdb
+rm -f /etc/tomcat/keystore
+rm -f /etc/pki/katello/keystore
+cp -r #{options[:ssl_dir]} /root/ssl-build.#{old_hostname}#{timestamp}
+rm -rf /root/ssl-build
```

Comment:
Seems like we are doing something wrong in the installer if oyu have to delete all the final certificates vs. changing them in the installer and it propagating the changes.